diff -urN Mail-1.2.0.orig/Mail/smtp.php Mail-1.2.0/Mail/smtp.php
--- Mail-1.2.0.orig/Mail/smtp.php	1970-01-01 18:14:39.000000000 +0900
+++ Mail-1.2.0/Mail/smtp.php	2012-06-05 16:24:09.000000000 +0900
@@ -346,7 +346,7 @@
         }
 
         include_once 'Net/SMTP.php';
-        $this->_smtp = &new Net_SMTP($this->host,
+        $this->_smtp = new Net_SMTP($this->host,
                                      $this->port,
                                      $this->localhost);
 
