--- a/plugins/node.d/mysql_.in
+++ b/plugins/node.d/mysql_.in
@@ -882,8 +882,11 @@ sub show {
 	if $graph_name =~ /innodb_/ && $data->{_innodb_disabled};
 
     for my $ds (@{$graph->{data_sources}}) {
-	printf "%s.value %s\n",
-	    clean_fieldname($ds->{name}), $data->{$ds->{name}};
+        my $value = exists $ds->{value}
+            ? $ds->{value}($data)
+                : $data->{$ds->{name}};
+
+        printf "%s.value %s\n", clean_fieldname($ds->{name}), defined($value) ? $value : 'U';
     }
 
     return 0;
@@ -1214,12 +1217,26 @@ sub parse_file_io {
 sub parse_insert_buffer_and_adaptive_hash_index {
     parse_section(
 	sub {
-	    m/\G(\d+) inserts, (\d+) merged recs, (\d+) merges\n/gc && do {
-		$data->{ib_ibuf_inserts}    = $1;
-		$data->{ib_ibuf_merged_rec} = $2;
-		$data->{ib_ibuf_merges}     = $3;
-		return 1;
-	    };
+            # MySQL < 5.5
+            m/\G(\d+) inserts, (\d+) merged recs, (\d+) merges\n/gc && do {
+                $data->{ib_ibuf_inserts}    = $1;
+                $data->{ib_ibuf_merged_rec} = $2;
+                $data->{ib_ibuf_merges}     = $3;
+                return 1;
+            };
+            # MySQL >= 5.5
+            m/\Gmerged operations:\n insert (\d+), delete mark \d+, delete \d+\ndiscarded operations:\n insert (\d+), delete mark \d+, delete \d+\n/gc && do {
+                $data->{ib_ibuf_inserts} = $1;
+                $data->{ib_ibuf_merged_rec} = $1 + $2;
+                return 1;
+            };
+            m/\GIbuf: size (\d+), free list len (\d+), seg size (\d+),(?: (\d+) merges)?\n/gc && do {
+                $data->{ib_ibuf_size}     = $1;
+                $data->{ib_ibuf_free_len} = $2;
+                $data->{ib_ibuf_seg_size} = $3;
+                $data->{ib_ibuf_merges}   = $4 if defined $4; # MySQL >= 5.5
+                return 1;
+            };
 	}
     );
 }
