--- ImageMagick-6.8.9-2-orig/magick/fourier.c	2014-02-23 07:12:00.000000000 +0900
+++ ImageMagick-6.8.9-2/magick/fourier.c	2014-06-04 09:12:48.635423560 +0900
@@ -64,7 +64,7 @@
 #include "magick/thread-private.h"
 #if defined(MAGICKCORE_FFTW_DELEGATE)
 #if defined(MAGICKCORE_HAVE_COMPLEX_H)
-#include <complex.h>
+#include "complex.h"
 #endif
 #include <fftw3.h>
 #if !defined(MAGICKCORE_HAVE_CABS)
