--- src/util/sys_defs.h.orig	2011-01-17 23:44:25.000000000 +0900
+++ src/util/sys_defs.h	2011-02-11 20:41:20.192945527 +0900
@@ -400,7 +400,7 @@
 #define DEF_DB_TYPE	"dbm"
 #define ALIAS_DB_MAP	"dbm:/etc/mail/aliases"
 #define HAS_NIS
-#define HAS_NISPLUS
+/* #define HAS_NISPLUS */
 #define USE_SYS_SOCKIO_H		/* Solaris 2.5, changed sys/ioctl.h */
 #define GETTIMEOFDAY(t)	gettimeofday(t)
 #define ROOT_PATH	"/bin:/usr/bin:/sbin:/usr/sbin:/usr/ucb"
